src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -219,6 +223,65 @@ object Bams extends LazyLogging {
     }
   }
 
+  /** Returns an iterator over the records in the given reader such that the order of the records returned is determined
+    * by the value of the given SAM tag, which can optionally be transformed.
+    *
+    * @param iterator an iterator from which to consume records
+    * @param header the header to use for the sorted records
+    * @param maxInMemory the maximum number of records to keep and sort in memory if sorting is needed
+    * @param tmpDir the temporary directory to use when spilling to disk
+    * @param tag the SAM tag (two-letter key) to sort by
+    * @param defaultValue the default value, if any, otherwise require all records to have the SAM tag.
+    * @param transform the transform to apply to the value of the SAM tags (default is the identity)
+    * @param ordering the ordering of [[B]].
+    * @tparam A the type of the SAM tag
+    * @tparam B the type of the SAM tag after any transformation, or just [[A]] if no transform is given.
+    * @return an Iterator over records sorted by the given SAM tag, optionally transformed.
+    */
+  def sortByTransformedTag[A, B](iterator: Iterator[SamRecord],
+                                 header: SAMFileHeader,
+                                 maxInMemory: Int = MaxInMemory,
+                                 tmpDir: DirPath = Io.tmpDir,
+                                 tag: String,
+                                 defaultValue: Option[A] = None,
+                                 transform: A => B)
+                                (implicit ordering: Ordering[B]): Iterator[SamRecord] = {
```

Comment:
I'm curious.  Is there a reason to prefer using this implicit vs. just having `B <: Ordered`?  I see this done elsewhere too, but I don't know why it's preferred.

Reply:
Just following on from this.... since you have the implicit here, can it be provided explicitly?  And if yes, could we collapse the two methods and do away with the transform?

I.e. could I do:

```scala
sortByTag[String](iter, header, "RX")((a,b) => a.reverse <= b.reverse)
```

Reply:
(1) `Int` doesn't implement `Ordered[Int]`.  Instead, you can require an implicit and in scala PreDef there is an implicit conversion from `Int` to `RichInt` which gives you `Ordered[Int]`.
(2) That _almost_ works, but doesn't, since `<=` relies on the type being ordered.